### PR TITLE
chore: update changelog for 0.3.1-rc

### LINKED ISSUE #478: chore: changelog update for 0.3.1-rc

### PR DESCRIPTION
**Pull request by esavier**
fixes #478

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,124 +1,116 @@
...
-### Tests
-Oraganized black-box testing
-Added automated tests for HELM deployment
+### New Components
+- Object Builder (OB)
+- Materializer - ondemand (M-OD)
+- Materializer - general (M-G)
+- CDL Web Admin (GUI)
+- Edge Registry - (ER)
+- Partial Update Engine - (PUA)
```
Comment (luke-biel) [inline CHANGELOG.md:12]: PUA?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,124 +1,116 @@
...
+- CDL now uses Open Telemetry
+- DataRouter supports Message Batching
+
+### General Changes:
+- Object Builder is available in Graphql API
+- ObjectBuilder supports empty filters
+- Object Builder can use filters in queries
+- Edge Registry is available in Graphql API
+- Edge Registry notifications support added
+- All services can be queried with --help to get full list of options with explanations
```
Comment (luke-biel) [inline CHANGELOG.md:26]: I dunno if that will stay after I add config-files. In first version I've removed any trace of clap.

File: CHANGELOG.md (hunk 1 above)
Comment (frondeus) [reply CHANGELOG.md:12]:
```suggestion
- Partial Update Engine - (PUE)
```

File: CHANGELOG.md (hunk 2 above)
Comment (esavier) [reply CHANGELOG.md:26]: yeah i remember but we will think about it when it gets merged

Comment (luke-biel) [reply CHANGELOG.md:26]: It got merged on Saturday

Comment (luke-biel) [reply CHANGELOG.md:26]: In theory it's not in `rc` as you've pushed tag before the merge, but for 0.3 `gold` it's pretty much in

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,124 +1,116 @@
...
+
+### Changes to existing components:
+- Schema Registry now uses Postgres Backend
+- CDL can now work in ordered (messages) mode
+- CDL now uses Open Telemetry
+- DataRouter supports Message Batching
+
+### General Changes:
+- Object Builder is available in Graphql API
+- ObjectBuilder supports empty filters
```
Comment (frondeus) [inline CHANGELOG.md:22]:
```suggestion
- Object Builder supports empty filters
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,124 +1,116 @@
...
+- Materializer - general (M-G)
+- CDL Web Admin (GUI)
+- Edge Registry - (ER)
+- Partial Update Engine - (PUE)
+
+### Changes to existing components:
+- Schema Registry now uses Postgres Backend
+- CDL can now work in ordered (messages) mode
+- CDL now uses Open Telemetry
+- DataRouter supports Message Batching
```
Comment (frondeus) [inline CHANGELOG.md:18]:
```suggestion
- Data Router supports Message Batching
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,124 +1,116 @@
...
+### General Changes:
+- Object Builder is available in Graphql API
+- ObjectBuilder supports empty filters
+- Object Builder can use filters in queries
+- Edge Registry is available in Graphql API
+- Edge Registry notifications support added
+- All services can be queried with --help to get full list of options with explanations
+
+### Major Bugfixes
+- DataRouter - It is now possible to swith to multi/single threaded behaviour
```
Comment (frondeus) [inline CHANGELOG.md:29]:
```suggestion
- Data Router - It is now possible to swith to multi/single threaded behaviour
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,124 +1,116 @@
...
+
+### Documentation:
+- General Materialization documentation and how-to
+- CDL versioning
+- Updated ENV variables and their descriptions
+- Add Feature List and descriptions
+- GRPC implementation
```
Comment (frondeus) [inline CHANGELOG.md:36]:
```suggestion
- gRPC implementation
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,124 +1,116 @@
...
+- Consumer-based services will log error and continue on invalid message. no longer resulting in crash loop
+- Query multiple returns error for timeseries
+- build script fix
+- Metrics port defaulted to invalid ones
+
+### RFCs:
+- Materialization RFC
+- RFC index page
+- partial materialization
+- edge repository
```
Comment (frondeus) [inline CHANGELOG.md:48]:
```suggestion
- Edge Registry
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,124 +1,116 @@
...
-Added RAW queries to postgres
-Added RAW to query-router
-Added VictoriaMetrics Support
-Added check if Kafka/AMQP topic exists before inserting new schema in schema registry
-Added GRPC alternative ingestion method to command-service
-Added input batching for TimeSeries
-Added GraphQL server - `api`
-Added (opt-in) message ordering
-Added input-output communication method selection from ENV vars and command-line arguments
+- Added store type of schema in schema registry
```
Comment (frondeus) [inline CHANGELOG.md:59]:
```suggestion
- Added store type of schema in Schema Registry
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,124 +1,116 @@
...
+- Added RAW queries to postgres
+- Added RAW to query-router
```
Comment (frondeus) [inline CHANGELOG.md:61]:
```suggestion
- Added RAW to Query Router
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,124 +1,116 @@
...
+- Added VictoriaMetrics Support
+- Added check if Kafka/AMQP topic exists before inserting new schema in schema registry
```
Comment (frondeus) [inline CHANGELOG.md:63]:
```suggestion
- Added check if Kafka/AMQP topic exists before inserting new schema in Schema Registry
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,124 +1,116 @@
...
+- Added GRPC alternative ingestion method to command-service
```
Comment (frondeus) [inline CHANGELOG.md:64]:
```suggestion
- Added gRPC alternative ingestion method to command-service
```

**Review by luke-biel — APPROVED** [review]

**Pull request merged**